state/subnets.go

```diff
@@ -252,42 +252,51 @@ func (s *Subnet) Update(args network.SubnetInfo) error {
 				return nil, errors.Trace(err)
 			}
 		}
-		makeSpaceNameUpdate, err := s.updateSpaceName(args.SpaceName)
+		return s.UpdateOps(args)
+	}
+	return errors.Trace(s.st.db().Run(buildTxn))
+}
+
+// UpdateOps returns the OPs to add new info to the subnet based on input info.
+// Currently no data is changed unless it is the "undefined" space from MAAS.
+// There are restrictions on the additions allowed:
+//   - No change to CIDR; more work is required to determine how to handle.
+//   - No change to ProviderId nor ProviderNetworkID; these are immutable.
+func (s *Subnet) UpdateOps(args network.SubnetInfo) ([]txn.Op, error) {
```

Comment:
Do we want to use this patch for updating the Subnet SpaceID?
If yes:
- need to change to support a move from a space other than `alpha` name -> what implication does that have for the existing code? It is used for the updating part from the provider update
-  Does a check for exiting key `FanLocalUnderlay` make sense for this patch?

Reply:
@manadart  ^

Reply:
Yes, we no longer restrict to moving from `alpha`.
And yes, we need to move fan overlay/underlay together.

Reply:
For now, I was planning to exclude the changes to that place, as the patch was getting bigger. 
d983087c9190e5e77b43920e4bb5f9d4f2c0c27b uses an extracted version of `getUpdateSpaceOps`, which only checks for the spaceName and omits the fan overlay/underlay check. Would that work? I expected that we don't need this, as we are not really touching the machines.

Reply:
If you look at `state.AddSpace` you can see that we never explicitly indicate a fan overlay space. It always inherits the space from its underlay.